### PR TITLE
Handle isFileSystemResource context key

### PR DESCRIPTION
**Pull request by dhuebner**
#### What it does

As mentioned [here](https://github.com/eclipse-theia/theia/issues/13535#issuecomment-2066587202) the handling for the `isFileSystemResource` is missing in Theia. This prevents menu contributions that are using this key.  

Context key description: https://code.visualstudio.com/api/references/when-clause-contexts#available-context-keys

#### How to test

Simplest way to test is to install the [Open In External App](https://open-vsx.org/extension/YuTengjing/open-in-external-app) extension as described in #13535. With this PR applied you should see the "Open in External App" Explorer context menu entry.
Bildschirmfoto 2024-04-29 um 15 15 37

#### Review checklist

- [ x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

File: packages/navigator/src/browser/navigator-widget.tsx [hunk 1]
```diff
@@ -210,6 +210,7 @@ export class FileNavigatorWidget extends AbstractNavigatorTreeWidget {
 
     protected updateSelectionContextKeys(): void {
         this.contextKeyService.explorerResourceIsFolder.set(DirNode.is(this.model.selectedNodes[0]));
+        this.contextKeyService.isFileSystemResource.set(FileStatNodeData.is(this.model.selectedNodes[0]));
```
Comment (msujew) [inline packages/navigator/src/browser/navigator-widget.tsx:213]: Suggestion: VSCode additionally tests whether the file service has a provider for the URI. See [here](https://github.com/microsoft/vscode/blob/1867e88862ca26343822231267f20b839daef65d/src/vs/workbench/common/contextkeys.ts#L254). We should likely do the same.

Comment (dhuebner) [reply packages/navigator/src/browser/navigator-widget.tsx:213]: @msujew 
Yes, this is true. If I'm not mistaken, in Theia this is indirectly done by FileSystemService that gives FileStats only to entires that has a FileSystemProvider for the given schema. So I skipped the additional provider check. However, I can add it here too if needed.

Comment (msujew) [reply packages/navigator/src/browser/navigator-widget.tsx:213]: Ok, that's fine. Do you mind adding a comment that explains this implicit behavior then?

Comment (dhuebner) [reply packages/navigator/src/browser/navigator-widget.tsx:216]: Comment is added

**Review by msujew — APPROVED** [review]
Thanks, LGTM 👍

**Pull request merged**